gensim/models/ldamodel.py

```diff
@@ -83,6 +83,87 @@ def logsumexp(x):
 
     return x
 
+def mean_absolute_difference(a, b):
+    """
+    Mean absolute difference between two arrays
+
+    Parameters
+    ----------
+    a : (M,) array_like of float32
+    b : (M,) array_like of float32
+
+    Returns
+    -------
+    float32
+        mean(abs(a - b))
+    
+    """
+    return np.mean(np.abs(a - b))
+
+try:
```

Comment:
not sure how to restructure this yet.  The except clause needs to happen after the "slow" versions of logsumexp and mean_absolute_difference are defined, so that's why the code is where it is right now.  I initially included the definition of the slow versions in the except block as well (mirroring what word2vec does), but that breaks if anyone uses the float16 dtype which requires the slow functions even if the cython versions exist (since the cython versions can only handle float32 and float 64).

might make things easier just to move the "inner" functions (logsumexp, dirichlet_expectation, and mean_abs_diff) into their own python module and have that module import the cython versions if available.  Open to any other suggestions as well.

Reply:
Ok, let's leave it for now as is